File: atcoder/abc044c.cc
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
ll dp[55][55][50 * 50 + 5];
int main(void)
{
    int n, a, x;
    cin >> n >> a;
    dp[0][0][0] = 1;
    for (int i = 1; i <= n; i++) {
        cin >> x;
        for (int j = 0; j <= i; j++) {
            for (ll sum = 0; sum <= 50 * i; sum++) {
                if (dp[i - 1][j][sum]) {
                    dp[i][j][sum] += dp[i - 1][j][sum];
                    dp[i][j + 1][sum + x] += dp[i - 1][j][sum];
                }
            }
        }
    }
    ll res = 0;
    for (int i = 1; i <= n; i++)
        res += dp[n][i][a * i];
    cout << res;
    return 0;
}

File: atcoder/abc044c.py
N, A = map(int, input().split())
X = list(map(int, input().split()))
from collections import defaultdict

# dp[i][j][k] == select j items in total i things, sum is k's number
dp = [[defaultdict(int) for _ in range(N + 2)] for _ in range(N + 2)]
dp[0][0][0] = 1

for i in range(1, N + 1):
    for j in range(0, i + 1):
        for k in range(50 * i + 1):
            if dp[i - 1][j][k]:
                dp[i][j][k] += dp[i - 1][j][k]
                dp[i][j + 1][k + X[i - 1]] += dp[i - 1][j][k]

res = 0
for i in range(1, N + 1):
    res += dp[N][i][A * i]

print(res)
